docs/upgrade/read-model-for-product-lists.md

```diff
@@ -0,0 +1,99 @@
+        ```
+    - in `ProductController::listByCategoryAction`:
+        ```diff
+        - $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsInCategory(
+        -      $productFilterData,
+        -      $orderingModeId,
+        -      $page,
+        -      self::PRODUCTS_PER_PAGE,
+        -      $id
+        -  );
+        + $paginationResult = $this->listedProductsFacade->getFilteredPaginatedInCategory(
+        +      $id,
+        +      $productFilterData,
+        +      $orderingModeId,
+        +      $page,
+        +      self::PRODUCTS_PER_PAGE
+        +  );
+        ```
+    - in `ProductController::searchAction`:
+        ```diff
+        - $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsForSearch(
+        + $paginationResult = $this->listedProductsFacade->getFilteredPaginatedForSearch(
+        ```
+    - in `ProductController::listByBrandAction`:
+        ```diff
+        - $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsForBrand(
+        -     $orderingModeId,
+        -     $page,
+        -     self::PRODUCTS_PER_PAGE,
+        -     $id
+        - );
+        + $paginationResult = $this->listedProductsFacade->getPaginatedForBrand(
+        +     $id,
+        +     $orderingModeId,
+        +     $page,
+        +     self::PRODUCTS_PER_PAGE
+        + );
+        ```
+    - in `CartController::addProductAjaxAxction`:
+        ```diff
+        - $accessories = $this->productAccessoryFacade->getTopOfferedAccessories(
+        -     $addProductResult->getCartItem()->getProduct(),
+        -     $this->domain->getId(),
+        -     $this->currentCustomer->getPricingGroup(),
+        -     self::AFTER_ADD_WINDOW_ACCESSORIES_LIMIT
+        );
+        + $accessories = $this->listedProductsFacade->getAccessories(
+        +     $addProductResult->getCartItem()->getProduct()->getId(),
+        +     self::AFTER_ADD_WINDOW_ACCESSORIES_LIMIT
+        + );
+        ```
+    - in `HomepageController::indexAction`:
+        ```diff
+        - $topProducts = $this->topProductFacade->getAllOfferedProducts(
+        -     $this->domain->getId(),
+        -     $this->currentCustomer->getPricingGroup()
+        - );
+        + $topProducts = $this->listedProductsFacade->getAllTop();
+        ```
+- edit your `productListMacro.html.twig` so it now works with instances of `ListedProductView` instead of `Product` entities
+    - to render product flags by their ids, you need to implement new [`FlagsExtension`](/project-base/src/Shopsys/ShopBundle/Twig/FlagsExtension.php) with `renderFlagsByIds` function and add a new [`productFlags.html.twig`](/project-base/src/Shopsys/ShopBundle/Resources/views/Front/Inline/Product/productFlags.html.twig) template
+    - to render "add to cart" form, add `CartController::productActionAction` and use it instead of `addProductFormAction`
```

Comment:
To render... add... what? And use it where?

If I have to implement something in the `CartController`, please provide a diff, it is super helpful.
And please change the "use it" instruction to "change this there, and change this there"